### ISSUE TITLE
cannot access elements in iframes when after_hook accesses the DOM

### ISSUE THREAD

**Issue by MatthewMazaika** [watir/watir#888]
## Meta -
Watir Version: 6.16.5
Selenium Version: 3.142.7
Browser Version: Chrome 79.0.3945.79
Browser Driver Version: Chromedriver 79.03945.36
OS Version: Linux Mint 19.1

## Expected Behavior -
calls to elements in an iframe work even if an `after_hook` is accessing the DOM

## Actual Behavior -
cannot find elements in an iframe

## Steps to reproduce -
slightly modifying the test [here](https://github.com/watir/watir/blob/284f1dd7fae52ee5d60804f80a30aab039ff4786/spec/watirspec/after_hooks_spec.rb#L116)  passes successfully when the hook accesses the browser's title
```
    it 'runs after_hooks after FramedDriver#switch!' do
      browser.goto(WatirSpec.url_for('iframes.html'))
      @page_after_hook = proc { @yield = browser.title == 'Iframes' }
      browser.after_hooks.add @page_after_hook

      expect(browser.h1.text).to eq 'Iframes'
      expect(browser.iframe.element(css: '#senderElement')).to exist

      expect(@yield).to be true
    end
```

but when changing the hook to look at an element in the DOM, we get an error that it cannot find the element in the iframe
```
    it 'runs after_hooks after FramedDriver#switch!' do
      browser.goto(WatirSpec.url_for('iframes.html'))
      @page_after_hook = proc { @yield = browser.h1.text == 'Iframes' }
      browser.after_hooks.add @page_after_hook

      expect(browser.h1.text).to eq 'Iframes'
      expect(browser.iframe.element(css: '#senderElement')).to exist

      expect(@yield).to be true
    end
```

```
     Failure/Error: expect(browser.iframe.element(css: '#senderElement')).to exist
       expected #<Watir::HTMLElement: located: false; {:tag_name=>"iframe"} --> {:css=>"#senderElement"}> to exist
```

**Comment by MatthewMazaika** [comment]
in `browser.ensure_context`, we flip to the default context and run the hook

i went ahead and took a quick pass at having it then flip back to the frame when the after_hook completed

i exposed a new property on the `Browser` called `current_frame` and set that in the iframe code
```
    def switch!
      @browser.current_frame = @element # <---- this is new
      @driver.switch_to.frame @element
      @browser.default_context = false
      @browser.after_hooks.run
    rescue Selenium::WebDriver::Error::NoSuchFrameError => e
      raise UnknownFrameException, e.message
    end
```

and then when we aren't in the default context, we flip back to the frame
```
    def ensure_context
      return if @default_context

      driver.switch_to.default_content
      @default_context = true
      after_hooks.run
      driver.switch_to.frame @current_frame
      @default_context = false
    end
```
it looks like it got my test to pass, but if i attempt to make a second call to an element inside the iframe, the second call fails :cry: 

i'm also about 99% sure that this behavior is now actually running the checker in _both_ contexts - the main frame _and_ the iframe - as a side effect, it breaks a handful of tests :sob:

I'm not sure if this is the change we are looking for... :thinking:

**Comment by MatthewMazaika** [comment]
i can get moving again if i revert https://github.com/watir/watir/pull/837, but not sure that's the change we're looking for either

**Comment by titusfortner** [comment]
So, just wanted to update this since I didn't work on a fix for the 6.17 release...
I really want to provide an easy way to specify which afterhooks get run where, but that's out of scope for a bit still...
Reverting #837 might make the most sense and then update nokogiri to monkeypatch the class.
